lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -135,7 +135,7 @@ export class AuthCache extends BrowserStorage {// Singleton
         for (key in storage) {
             if (!state || key.indexOf(state) !== -1) {
                 const splitKey = key.split(Constants.resourceDelimiter);
```

Comment:
Would it be simpler to just do key.endsWith(state) check instead of doing splits? state always has Guid which should prevent any accidental matches. If you are concerned with older browsers then you already have UrlUtils.endsWith() method

Reply:
Good suggestion. May be that works too .. we wanted to add more params for state and hence did it this way, @jasonnutter  can you please follow up?

Reply:
Okay, we thought about it and I do not want to use key.endsWith(state) as this will only validate if the param we are checking `against` has state but not the exact structure we expect it to have. That would be an incomplete check - necessary but not sufficient (even if it is simple). 

Hence I am leaning towards the original solution for now.

Reply:
Turns out, we should just check if the state string is present in the string, as there shouldn't be a scenario where the state string is in the key but not at the end. Making this logic not dependent on the resource delimiter both solves this bug, but is also future proof if we want to change the format of the state string (which we are considering).